src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.NormalPriorityProcessor.cs

```diff
@@ -438,37 +438,32 @@ static async Task DocumentCloseAsync(IIncrementalAnalyzer analyzer, TextDocument
                         }
                     }
 
-                    private async Task ProcessReanalyzeDocumentAsync(WorkItem workItem, TextDocument document, CancellationToken cancellationToken)
+                    private async Task ProcessReanalyzeDocumentAsync(ImmutableArray<IIncrementalAnalyzer> analyzers, WorkItem workItem, TextDocument document, CancellationToken cancellationToken)
                     {
                         try
                         {
-#if DEBUG
-                            Debug.Assert(!workItem.InvocationReasons.Contains(PredefinedInvocationReasons.Reanalyze) || workItem.SpecificAnalyzers.Count > 0);
-#endif
-
                             // No-reanalyze request or we already have a request to re-analyze every thing
                             if (workItem.MustRefresh || !workItem.InvocationReasons.Contains(PredefinedInvocationReasons.Reanalyze))
                             {
                                 return;
                             }
 
                             // First reset the document state in analyzers.
-                            var reanalyzers = workItem.SpecificAnalyzers.ToImmutableArray();
```

Comment:
Just from looking at the code, and having no other context, I'm wondering if perhaps this method was intended to run only when there are specific analyzers on a work item.. in which case perhaps rather than removing the assert, it would make more sense to simply add `|| workItem.SpecificAnalyzers.Count > 0` to the early-return guard above?

But at the same time, what I have here also makes sense. All of the tests pass too, so I really don't know what exactly to go on, to try to work out what this method is intended to do.

Reply:
Yeah, the core issue seems to be "Reanalyze" was designed only for a single/subset of specific analyzers. I think your change is fine to unblock us, and lets pray that @CyrusNajmabadi kills the solution crawler before this change uncovers more bugs :-)

Reply:
So I shouldn't point out the bug I found, where if you call `item.With(..., ..., item.AsyncToken)` the first thing the `With` method does is dispose the async token that you're trying to re-use? :D

Reply:
Doesn't it dispose the old one and re-use the new one?
https://github.com/dotnet/roslyn/blob/d04c23b6a71cf74df4131b2e0be1184873bb4db1/src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.WorkItem.cs#L133-L143

Reply:
Right, bit there are spots where the old one is passed in, to be the new one, presumably assuming it will just be reused.

Reply:
Oops

Reply:
Probably add a guard that AsyncToken != asyncToken around the above Dispose call?

Reply:
I don't know if I want to touch it.. this is going into 17.4 under QB and I don't understand the implications